static/js/protein/chart.ts

```diff
@@ -242,34 +258,117 @@ const LINK_STYLE = {
   },
 };
 
-// https://stackoverflow.com/a/69610045
-function brighten(): void {
-  // brighten object under cursor
-  d3.select(this).style("filter", `brightness(${BRIGHTEN_PERCENTAGE})`);
+/**
```

Comment:
nit: I think all of these helper functions: brighten, unbrighten, updateToolTipText, updateToolTipPosition, showTooltip and hideTooltip are not necessary because:
1. they're each only used in one place
2. they all just pass parameters to one other function and don't really contain any logic

Reply:
Makes sense, sounds good!